storage/tianmu/core/tianmu_attr_exqp.cpp

```diff
@@ -875,19 +875,29 @@ void TianmuAttr::EvaluatePack_BetweenInt(MIUpdatingIterator &mit, int dim, Descr
     mit.NextPackrow();
     return;
   }
-  int64_t pv1 = d.val1.vc->GetValueInt64(mit);
-  int64_t pv2 = d.val2.vc->GetValueInt64(mit);
+  int64_t pv1 = d.val1.vc->GetValueInt64(mit);  // e.g.: between pv1 and pv2
+  int64_t pv2 = d.val2.vc->GetValueInt64(mit);  // e.g.: between pv1 and pv2
+  uint64_t upv1, upv2;                          // used uint64_t to store level 2 encode, ranges from [0, uint64_max]
+  // Only used when rough check = RS_ALL(no numeric data matchs), but table has null, rough check will be reset RS_SOME,
+  // so function EvaluatePack_BetweenInt() still be called to filter nulls, but no data matchs. This is a tmp solution,
+  // we may find a better way to deal with this case. Used with upv2.
+  bool empty_flag = true;
   int64_t local_min = dpn.min_i;
   int64_t local_max = dpn.max_i;
-  if (pv1 != common::MINUS_INF_64)
-    pv1 = pv1 - local_min;
+  if (common::MINUS_INF_64 == pv1 || pv1 <= local_min)
+    // we reserve common::MINUS_INF_64 case for "where a < 3" tranform to "[MINUS_INF_64, 2]".
+    upv1 = 0;
   else
-    pv1 = 0;
+    upv1 = pv1 - local_min;
```

Comment:
and how about  local_min is MIN and pv1 is MAX ?  `select xxxx where  xx between MIN and MAX`